engine.py

```diff
@@ -476,16 +478,66 @@ def execute(cls, ids, data):
 
             if action.html_copies and action.html_copies > 1:
                 content = cls.merge_pdfs([content] * action.html_copies)
-
             Printer = None
             try:
                 Printer = Pool().get('printer')
             except KeyError:
                 logger.warning('Model "Printer" not found.')
             if Printer:
-                return Printer.send_report(oext, content,
+                result = Printer.send_report(oext, content,
                     action_name, action)
-        return oext, content, cls.get_direct_print(action), filename
+                if result:
+                    oext, content, direct_print, filename = result
+
+        if queue and content:
+            with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+                temp_file.write(content)
+            try:
+                queue.put((oext, temp_file.name, cls.get_direct_print(action), filename))
+            except Exception as e:
+                queue.put(e)
+        elif content:
+            return (oext, content, cls.get_direct_print(action), filename)
+
+    @classmethod
+    def execute(cls, ids, data):
+        cls.check_access()
+        action, model = cls.get_action(data)
+        # in case is not jinja, call super()
+        if action.template_extension != 'jinja':
+            return super().execute(ids, data)
+
+        context = Transaction().context
+        if context.get('timeout'):
+            timeout = context['timeout']
+
+            queue = multiprocessing.SimpleQueue()
+            process = multiprocessing.Process(target=cls.__execute, args=(ids, data, queue,))
+            process.start()
+
+            try:
+                process.join(timeout=timeout)
+
+                if process.is_alive():
+                    process.terminate()
+                    raise UserError(gettext('html_report.msg_error_timeout'))
+                else:
+                    if not queue.empty():
```

Comment:
I think that if queue is empty we should raise an UserError

Reply:
Done at bdd8a88fb413f221b53fff7b2f7dda982e219ecd